erts/preloaded/src/prim_file.erl

```diff
@@ -483,22 +491,30 @@ get_fd_data(#file_descriptor{ data = Data }) ->
         _ -> error(not_on_controlling_process)
     end.
 
+get_fd_data_for_read(Fd) ->
+    #{ r_buffer := RBuf} = Data = get_fd_data(Fd),
+    case RBuf of
+        write -> throw({error, ebadf});
+        _ -> Data
+    end.
+
+
 build_fd_data(FRef, Modes) ->
     Defaults =
         #{ owner => self(),
            handle => FRef,
            r_ahead_size => 0,
-           r_buffer => prim_buffer:new() },
+           r_buffer => write },
```

Comment:
Sorry about the delay in getting back to you. This breaks `prim_file_SUITE:modes/1`, the default is `read` when `write` is not given, causing the part below to fail:

```
  425:     %% read by default
  426:     {ok, Fd4} = ?PRIM_FILE:open(Name1, []),
  427:     {ok, Marker} = ?PRIM_FILE:read(Fd4, Length),
  428:     ok = ?PRIM_FILE:close(Fd4),
```